File: frontend/angular-book-ecommerce/src/app/common/cart-item.ts
import { Book } from "./book";

export class CartItem {
    id: number;
    title: string;
    thumbnail: string;
    unitPrice: number;
    
    quantity: number;
    
    constructor(book: Book){
        this.id = book.id;
        this.title = book.title;
        this.thumbnail= book.thumbnail;
        this.unitPrice = book.unitPrice;
    
        this.quantity = 1;
    }
}

File: backend/spring-boot-book-ecommerce/src/main/java/com/luvannie/springbootbookecommerce/entity/CartItem.java
package com.luvannie.springbootbookecommerce.entity;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "cart_item")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(
        generator = ObjectIdGenerators.PropertyGenerator.class,
        property = "id")
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "thumbnail")
    private String thumbnail;

    @Column(name = "unit_price")
    private double unitPrice;

    @Column(name = "quantity")
    private int quantity;


    @Column(name = "book_id")
    private Long bookId;


    @Column(name = "user_id")
    private Long userId;
}
